docs/tutorial/plot_function_transformer.py

```diff
@@ -0,0 +1,206 @@
+from onnxruntime import InferenceSession
+
+
+def calculate(df):
+    df["c"] = 100 * (df["a"] - df["b"]) / df["b"]
+    return df
+
+
+mapper = ColumnTransformer(
+    transformers=[
+        ("c", FunctionTransformer(calculate), ["a", "b"]),
+    ],
+    remainder="passthrough",
+    verbose_feature_names_out=False,
+)
+mapper.set_output(transform="pandas")
+
+pipe = Pipeline([("mapper", mapper), ("classifier", DecisionTreeClassifier())])
+
+data = DataFrame(
+    [
+        dict(a=1, b=2, f=5),
+        dict(a=4, b=5, f=10),
+    ]
+)
+y = np.array([0, 1], dtype=np.int64)
+pipe.fit(data, y)
+
+try:
+    to_onnx(pipe, data[:1], options={"zipmap": False})
+except Exception as e:
+    print("It does not work:", e)
+
+##################################
+# Use of custom transformer
+# +++++++++++++++++++++++++
+#
+# It is easier to write a custom converter if the FunctionTransformer
+# is implemented as a custom transformer.
+
+
+class OverpriceCalculator(BaseEstimator, TransformerMixin):
+    def __init__(self):
+        pass
+
+    def calculate_overprice(self, x, y):
+        return 100 * (x - y) / y
+
+    def fit(self, X, y=None):
+        return self
+
+    def transform(self, X, y=None):
+        x = X.apply(lambda x: self.calculate_overprice(x.a, x.b), axis=1)
+        return x.values.reshape((-1, 1))
+
+
+mapper = ColumnTransformer(
+    transformers=[
+        ("ab", FunctionTransformer(), ["a", "b"]),  # We keep the first column.
+        ("c", OverpriceCalculator(), ["a", "b"]),  # We add a new one.
+    ],
+    remainder="passthrough",
+    verbose_feature_names_out=False,
+)
+
+pipe_tr = Pipeline([("mapper", mapper), ("classifier", DecisionTreeClassifier())])
+pipe_tr.fit(data, y)
+
+#############################
+# Both pipelines return the same output.
+assert_allclose(pipe.predict_proba(data), pipe_tr.predict_proba(data))
+
+#############################
+# Let's check it produces the same number of features.
+assert_allclose(pipe.steps[0][-1].transform(data), pipe_tr.steps[0][-1].transform(data))
+
+#############################
+# But the conversion still fails with a different error message.
+
+try:
+    to_onnx(pipe_tr, data[:1], options={"zipmap": False})
+except Exception as e:
+    print("It does not work:", e)
+
+
+#################################
+# Custom converter
+# ++++++++++++++++
+#
+# We need to implement the method `calculate_overprice` in ONNX.
+# The first function returns the expected type and shape.
+
+
+def overprice_shape_calculator(operator):
+    check_input_and_output_numbers(operator, input_count_range=1, output_count_range=1)
+    # Gets the input type, the transformer works on any numerical type.
+    input_type = operator.inputs[0].type.__class__
+    # The first dimension is usually dynamic (batch dimension).
+    input_dim = operator.inputs[0].get_first_dimension()
+    operator.outputs[0].type = input_type([input_dim, 1])
+
+
+def overprice_converter(scope, operator, container):
```

Comment:
could you also please replace all occurrences of the word "overprice" with something else? I forgot to rename it while posting